pkg/scheduler/core/generic_scheduler.go

```diff
@@ -564,7 +576,11 @@ func podFitsOnNode(
 			//TODO (yastij) : compute average predicate restrictiveness to export it as Prometheus metric
 			if predicate, exist := predicateFuncs[predicateKey]; exist {
 				if eCacheAvailable {
-					fit, reasons, err = nodeCache.RunPredicate(predicate, predicateKey, pod, metaToUse, nodeInfoToUse, equivClass, cache)
+					predicateGeneration := uint64(0)
+					if predicateGenerations != nil {
```

Comment:
Actually, it's perfectly valid to do a lookup on a nil map. You will get the zero value.

Reply:
Updated, and put it into `nodeCache.RunPredicate(...)`.